templates/funcs.go

```diff
@@ -221,11 +233,31 @@ func render(f pgs.Field) string {
 
 	var s string
 
+	var obsType zap.ObfuscationType
+	if _, err := f.Extension(zap.E_ObfuscationType, &obsType); err != nil {
+		fmt.Fprintf(os.Stderr, "error getting obfuscation_type for field %s: %s", f.Name(), err)
```

Comment:
should the log state that we're falling back to HIDE for safety?

Reply:
Doc says
```
An error will only be returned if there is a type mismatch between desc and ext.
```
So between zap.E_ObfuscationType and obsType in this case

The error is not going to happen unless someone changes it without understanding it

Reply:
I know, but in case we mess it up later on (or get into a bug on a new protoc version), we'll know why fields are disappearing ;)

Reply:
It seems more dangerous to have no information rather than all of it, no ?

Reply:
If we rely on this for compliance with data protection rules, it's better to fail closed (default to hide, the current behaviour), than fail open (log them) and ask google to delete all of our logs.

Reply:
Alright, in this case why not panic here ? So to prevent this from happening at compilation before it hits prod

Reply:
Good point, this code runs during the code generation, and not on prod. Let's indeed panic to fail the release !